homeassistant/components/totalconnect/binary_sensor.py

```diff
@@ -0,0 +1,101 @@
+"""Interfaces with TotalConnect sensors."""
+import logging
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_DOOR,
+    DEVICE_CLASS_GAS,
+    DEVICE_CLASS_SMOKE,
+    BinarySensorDevice,
+)
+
+from . import DOMAIN as TOTALCONNECT_DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up a sensor for a TotalConnect device."""
+    if discovery_info is None:
+        return
+
+    sensors = []
+
+    client_locations = hass.data[TOTALCONNECT_DOMAIN].client.locations
+
+    for location_id in client_locations:
+        for zone in client_locations[location_id].zones:
```

Comment:
Since you are using the zone_id and passing it to TotalConnectBinarySensor.
```suggestion
        for zone_id in client_locations[location_id].zones:
```

Reply:
concur